spv/sync.go

```diff
@@ -540,6 +545,50 @@ func (s *Syncer) pickRemote(pick func(*p2p.RemotePeer) bool) (*p2p.RemotePeer, e
 	return nil, errors.E(errors.NoPeers)
 }
 
+// waitForAnyRemote blocks until there is one or more remote peers available or
+// the context is canceled.
+//
+// The [isEligible] callback determines which peers are eligible for selection.
+//
+// This function may return a nil peer with nil error if there are peers
+// available but none of them are eligible.
+//
+// If waitEligible is true, then this function will wait until at least one
+// remote is eligible.  Otherwise, this function returns a nil peer with nil
+// error when there are peers but none of them are eligible.
+func (s *Syncer) waitForRemote(ctx context.Context, isEligible func(rp *p2p.RemotePeer) bool, waitEligible bool) (*p2p.RemotePeer, error) {
```

Comment:
is there a reason to return `nil, nil` at all? (are there going to be future calls where `waitEligible` is false?)

this just seems super awkward to me, even more so because of the naming (`waitForRemote` might not wait at all for such a remote peer).

Reply:
if such a case is needed, i think using a different function (with a name to describe that non-waiting behavior) would be preferred.

Reply:
Yes, there will be a future commit where I rely on false and (nil, nil). Specifically, I'll use a picker to determine there are no peers with announced headers > than the currently synced to, to flag that the startup sync has finished.

I could also return nil, ErrNoPeer in this case and select on the error if that's preferable.